File: Algorithms/C++/Hash_Table/217_Contains_Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        if (nums.empty()) {
            return false;
        }

        unordered_set<int> hash;

        for (int i = 0; i < nums.size(); i++) {
            if (hash.find(nums[i]) == hash.end()) {
                hash.insert(nums[i]);
            } else {
                return true;
            }
        }

        return false;
    }
};

File: Algorithms/Swift/Easy/217_Contains_Duplicate.swift
class Solution {
    func containsDuplicate(nums: [Int]) -> Bool {
        if nums.isEmpty {
            return false
        }

        var dict = [Int : Int]()
        for num in nums {
            if let count = dict[num] {
                return true
            } else {
                dict[num] = 1
            }
        }

        return false
    }
}
